### PR TITLE
when determining the response to use, use the request's body with url…

### PR DESCRIPTION
**Pull request by edwardlafoy**
… and method

### CODE REVIEW

File: VCRURLConnection/VCRRecording.m [hunk 1]
```diff
@@ -69,7 +72,7 @@ - (NSUInteger)hash {
     NSUInteger hash = 1;
     hash = prime * hash + [self.method hash];
     hash = prime * hash + [self.URI hash];
-    hash = prime * hash + [self.body hash];
+    hash = prime * hash + [self.requestBodyData hash];
```
Comment (danhodos) [inline VCRURLConnection/VCRRecording.m]: What is `hash` used for? Should we be hashing on the `responseBodyData`, too?

File: VCRURLConnection/VCRRecording.m [hunk 2]
```diff
@@ -86,22 +89,41 @@ - (BOOL)isText {
...
+        return [[NSString alloc] initWithData:self.responseBodyData encoding:NSUTF8StringEncoding];
     } else {
-        return [self.data base64Encoding];
+        return [self.responseBodyData base64Encoding];
+    }
+}
+
+- (void)setRequestBody:(NSString * _Nullable)requestBody
+{
+    if ([requestBody isKindOfClass:[NSDictionary class]]) {
```
Comment (danhodos) [inline VCRURLConnection/VCRRecording.m]: This looks to be highly duplicated from `setResponseBody`, can we extract a method here, perhaps?

File: VCRURLConnection/VCRRecording.m [hunk 3]
```diff
@@ -86,22 +89,41 @@ - (BOOL)isText {
...
+        self.requestBodyData = [NSJSONSerialization dataWithJSONObject:requestBody options:0 error:nil];
+    } else if ([self isText]) {
+        self.requestBodyData = [requestBody dataUsingEncoding:NSUTF8StringEncoding];
+    } else if ([requestBody isKindOfClass:[NSString class]]) {
+        self.requestBodyData = [[NSData alloc] initWithBase64Encoding:requestBody];
+    }
+}
+
+- (NSString *)requestBody {
```
Comment (danhodos) [inline VCRURLConnection/VCRRecording.m]: Same between this and `responseBody`.

**Comment by danhodos** [comment]
Had some small comments about duplication, but otherwise it seems pretty rad. Yay open source software!

File: VCRURLConnection/VCRRecording.m (hunk 1 above)
Comment (edwardlafoy) [inline VCRURLConnection/VCRRecording.m]: The hash is used for the keys in `NSDictionary` in this instance, it's part of `NSObjectProtocol`:

https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Protocols/NSObject_Protocol/#//apple_ref/occ/intfm/NSObject/hash

Comment (edwardlafoy) [inline VCRURLConnection/VCRRecording.m]: the response body isn't part of what makes a unique request, so it's not included here.

File: VCRURLConnection/VCRRecording.m (hunk 3 above)
Comment (edwardlafoy) [inline VCRURLConnection/VCRRecording.m]: good call

**Pull request merged**